File: android/src/main/java/com/sicpa/didcomm/reactnative/ResolverProxyEvent.kt
package com.sicpa.didcomm.reactnative

sealed class ResolverProxyEvent(val type: String)

class ResolveDid(val did: String) : ResolverProxyEvent("resolve-did")
class FindKey(val kid: String) : ResolverProxyEvent("find-key")
class FindKeys(val kids: List<String>) : ResolverProxyEvent("find-keys")

File: ios/Source/ResolverProxyEvent.swift
import DidcommSDK
import Foundation

enum ResolverProxyEvent {
    case ResolveDid(did: String)
    case FindKey(kid: String)
    case FindKeys(kids: [String])
    
    var key: String {
        switch self {
        case .ResolveDid:
            return "resolve-did"
        case .FindKey:
            return "find-key"
        case .FindKeys:
            return "find-keys"
        }
    }
}
